openml/flows/sklearn_converter.py

```diff
@@ -32,37 +33,52 @@
     '^(?P<name>[\w\-]+)((?P<operation>==|>=|>)(?P<version>(\d+\.)?(\d+\.)?(\d+)))?$')
 
 
-def sklearn_to_flow(o):
+def sklearn_to_flow(o, parent_model=None):
+    # TODO: assert that only on first recursion lvl `parent_model` can be None
 
     if _is_estimator(o):
+        # is the main model or a submodel
         rval = _serialize_model(o)
     elif isinstance(o, (list, tuple)):
-        rval = [sklearn_to_flow(element) for element in o]
+        # TODO: explain what type of parameter is here
+        rval = [sklearn_to_flow(element, parent_model) for element in o]
         if isinstance(o, tuple):
+            assert(parent_model is not None)
```

Comment:
I think this checks should live in `_extract_information_from_model` as this function only distributes objects to the right serialization functions.